tests/virt_autotest/validate_host_and_guests.pm

```diff
@@ -0,0 +1,48 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: qe-virt@suse.de
+
+package validate_host_and_guests;
+
+use strict;
+use warnings;
+use base "virt_feature_test_base";
+use testapi;
+use virt_autotest::common;
+use virt_autotest::utils qw(check_host_health check_guest_health);
+
+sub run_test {
+
+    my $failures = check_host_health;
+    die "Host is not healthy: $failures" unless $failures eq "";
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        $failures = check_guest_health($guest);
+        die "Guest $guest is not healthy: $failures" unless $failures eq "";
+    }
+}
+
+sub prepare_run_test {
+    script_run("rm -f /root/{commands_history,commands_failure}");
+    assert_script_run("history -c");
+}
+
+sub post_fail_hook {
+    my $self = shift;
+    collect_virt_system_logs();
+    virt_utils::collect_host_and_guest_logs;
+    upload_logs("/var/log/clean_up_virt_logs.log");
+    upload_logs("/var/log/guest_console_monitor.log");
+    script_run("rm -f -r /var/log/clean_up_virt_logs.log /var/log/guest_console_monitor.log");
+    $self->upload_coredumps;
+    save_screenshot;
+}
+
+sub test_flags {
+    return {fatal => 1};
+}
+
+1;
```

Comment:
1. If you already put all checks in base class, you do not need this module anymore
2. I do not think all our test modules can guarantee all guests are in intact state after testing, because not all of them checking ssh connection. They just test their specific checkpoints. I am afraid test will spin around here for a while under certain circumstances. This will impact test run a lot sometimes.
3. If you insist on adding such module, you can add it conditionally, namely a parameter to control whether this module should be added in. And I do not think this module should be added in all test suites, this will lead to too many duplicate checkings.

Reply:
Anyway this module needs thorough testing, otherwise it can not be added into our test suites. IMHO.

Reply:
1. it's the resolution from our discussion yesterday. So I added it.
2. No, I am not sure either. The commit is just a draft, I need some runs to see the result.
3. I don't insist on adding this module. Let's see the verify result, then team members decide if it is required.

Reply:
Yes. Go and do thorough testing. And this module should be added conditionally if it is going to be added. Namely a parameter to control whether this module should be added in. And I do not think this module should be added in all test suites, this will lead to too many duplicate checkings.